Sources/Db/APIs/MySQL.php

```diff
@@ -2766,6 +2784,9 @@ protected function replacement__callback(array $matches, array $db_values, objec
 				$this->error_backtrace('Undefined type used in the database query. (' . $matches[1] . ':' . $matches[2] . ')', '', false, __FILE__, __LINE__);
 				break;
 		}
+
+		// We reached a impossible location, but static anlaysis doesnt know that.
+		throw new \Exception();
```

Comment:
```suggestion
		return '';
```

This will match SMF\Db\APIs\PostgreSQL::replacement__callback()

Reply:
We don't need that.  Static analysis thinks that we don't return in all instances, because it doesn't understand that the switch is the last code in the function and will ensure we do something.
The throw helps it understand we will not be executing beyond that point, and would stop executing code if we reach it as a fatal error.

Reply:
Then please also change the PostgreSQL file to match this. I don't really care which one we use, just as long as we do it consistently.

Reply:
My bad for missing that, done.

I did look at using the match statement, but I don't see it improving anything.  It would involve a try/catch block and a significant return on the match (`return match[....]`) and anonymous functions to handle more complex checks we do.  Didn't seem like a win just to help static analysis know.